src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -481,9 +495,77 @@ private void UpdateCustomColumnsVisibility(ImmutableDictionary<string, Immutable
+                        if (customColumnDefinition == null)
+                        {
+                            Debug.Fail($"{nameof(SourceReferenceItem.ReferenceUsageInfo)} has a key '{key}', but there is no exported '{nameof(AbstractCustomColumnDefinition)}' with this name.");
+                        }
+
+                        // Ensure that we flip the visibility to true for the custom column.
+                        // Note that the actual UI update happens outside the lock when we
+                        // invoke "TableControl.SetColumnStates" below.
+                        ColumnState2 newColumnStateOpt = null;
+                        if (_customColumnTitleToStatesMap.TryGetValue(customColumnDefinition.Name, out var columnState))
+                        {
+                            if (!columnState.IsVisible)
+                            {
+                                newColumnStateOpt = new ColumnState2(columnState.Name, isVisible: true, columnState.Width,
+                                    columnState.SortPriority, columnState.DescendingSort, columnState.GroupingPriority);
+                            }
+                        }
+                        else
+                        {
+                            newColumnStateOpt = customColumnDefinition.DefaultColumnState;
+                        }
+
+                        if (newColumnStateOpt != null)
+                        {
+                            _customColumnTitleToStatesMap[customColumnDefinition.Name] = newColumnStateOpt;
+
+                            newColumnStates.Add(newColumnStateOpt);
+                        }
+                    }
+
+                    // Update the column states if required.
+                    if (newColumnStates.Count > 0)
+                    {
+                        // SetColumnStates API forces a switch to UI thread, so it should be safe to call
+                        // from a background thread here.
+                        // Also note that we will call it only once for each new custom column to add for
+                        // each find references query - the lock above guarantees that newColumnStatesOpt is
+                        // going to be non-null only for the first result that has a non-empty column value.
+                        TableControl.SetColumnStates(newColumnStates.ToImmutable());
+                    }
+                }
+                finally
+                {
+                    newColumnStates.Free();
+                }
+            }
+
             public sealed override Task OnReferenceFoundAsync(SourceReferenceItem reference)
             {
-                UpdateCustomColumnsVisibility(reference.ReferenceInfo);
+                UpdateUsageColumnVisibility(reference.ReferenceUsageInfo);
+                UpdateCustomColumnVisibility("ContainingTypeInfo");
+                UpdateCustomColumnVisibility("ContainingMemberInfo");
```

Comment:
Should these use the referenced `ColumnDefinition.ColumnName` instead of strings?